router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -0,0 +1,325 @@
+package marketobulkupload
+
+import (
+	"bufio"
+	"encoding/json"
+	stdjson "encoding/json"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	time "time"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+type MarketoBulkUploader struct {
+	destName          string
+	destinationConfig map[string]interface{}
+	TransformUrl      string
+	PollUrl           string
+	logger            logger.Logger
+}
+
+func NewManager(destination *backendconfig.DestinationT) (*MarketoBulkUploader, error) {
+	marketoBulkUpload := &MarketoBulkUploader{destName: "MARKETO_BULK_UPLOAD", destinationConfig: destination.DestinationDefinition.Config, PollUrl: "/pollStatus", TransformUrl: config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090")}
+	return marketoBulkUpload, nil
+}
+
+var pkgLogger logger.Logger
+
+func init() {
+	pkgLogger = logger.NewLogger().Child("asyncdestinationmanager").Child("marketobulkupload")
+}
+
+func (b *MarketoBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
+	payload, err := json.Marshal(pollInput)
+	if err != nil {
+		// needs to be retried
+		// transformerConnectionStatus := 500
+		b.logger.Error("Error in Marshalling Poll Input: %w", err)
+		return common.PollStatusResponse{
+			StatusCode: 500,
+			HasFailed:  true,
+		}
+	}
+	bodyBytes, transformerConnectionStatus := misc.HTTPCallWithRetryWithTimeout(b.TransformUrl+b.PollUrl, payload, config.GetDuration("HttpClient.marketoBulkUpload.timeout", 30, time.Second))
+	if transformerConnectionStatus != 200 {
+		return common.PollStatusResponse{
+			StatusCode: transformerConnectionStatus,
+			HasFailed:  true,
+		}
+	}
+	var asyncResponse common.PollStatusResponse
+	err = json.Unmarshal(bodyBytes, &asyncResponse)
+	if err != nil {
+		// needs to be retried
+		// transformerConnectionStatus := 500
+		b.logger.Error("Error in Unmarshalling Poll Response: %w", err)
```

Comment:
Errorf?

Reply:
we are not returning error here, we are returning a response with 500